js/views/accountsettings.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * ownCloud - Mail
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Tahaa Karim <tahaalibra@gmail.com>
+ * @copyright Tahaa Karim 2016
+ */
+
+define(function(require) {
+	'use strict';
+
+	var $ = require('jquery');
+	var Marionette = require('marionette');
+	var Handlebars = require('handlebars');
+	var AccountSettingsTemplate = require('text!templates/accountsettings.html');
+	var AliasesView = require('views/aliases');
+	var Radio = require('radio');
+
+	return Marionette.LayoutView.extend({
+		template: Handlebars.compile(AccountSettingsTemplate),
+		templateHelpers: function() {
+			var aliases = this.aliases;
+			return {
+				aliases: aliases,
+				email: this.currentAccount.get('email')
+			};
+		},
+		currentAccount: null,
+		aliases: null,
+		ui: {
+			'form': 'form',
+			'alias': 'input[name="alias"]',
+			'submitButton': 'input[type=submit]',
+			'aliasName' : 'input[name="alias-name"]'
+		},
+		events: {
+			'click @ui.submitButton': 'onSubmit',
+			'submit @ui.form': 'onSubmit'
+		},
+		regions: {
+			aliasesRegion : '#aliases-list'
+		},
+		initialize: function(options) {
+			this.currentAccount = options.account;
+		},
+		onSubmit: function(e) {
+			e.preventDefault();
+			var alias = {
+				'alias': this.ui.alias.val(),
+				'name': this.ui.aliasName.val()
+			};
+			this.ui.alias.prop('disabled', true);
+			this.ui.aliasName.prop('disabled', true);
+			this.ui.submitButton.val('Saving');
+			this.ui.submitButton.prop('disabled', true);
+			var _this = this;
+
+			var savingAlias = Radio.aliases.trigger('save', this.currentAccount, alias);
+
+			$.when(savingAlias).done(function(alias) {
+				var aliases = [];
+				aliases.push(alias);
+				var json = require('state').accounts.get(_this.currentAccount.get('accountId')).toJSON();
+				for (var x in json.aliases) {
+					aliases.push(json.aliases[x]);
+				}
+				require('state').accounts.get(_this.currentAccount.get('accountId')).set({'aliases': aliases});
+				_this.currentAccount = require('state').accounts.get(_this.currentAccount.get('accountId'));
+				Radio.aliases.trigger('alias:add', aliases);
```

Comment:
while trying to solve the bug you mentioned on IRC I noticed that you're passing an array of aliases instead of the alias the was actually added. Is that intended?